ESP8266/ESP8266.h

```diff
@@ -21,6 +21,20 @@
 #include "nsapi_types.h"
 #include "rtos.h"
 
+// Various timeouts for different ESP8266 operations
+#ifndef ESP8266_CONNECT_TIMEOUT
+#define ESP8266_CONNECT_TIMEOUT 15000
```

Comment:
From what we've seen, TCP connections are an issue. This should be long enough to cover the longest TCP connection timeout in the chip. May be that you need a separate one for TCP and UDP.

Reply:
I'll check